gardener/util.go

```diff
@@ -0,0 +1,175 @@
+package gardener

```

Comment:
Please avoid names like util, common, helpers, etc... It is an antipattern.

Think about what that file contains. In this case it seems it is mostly `shoot` stuff, so it could be called `shoot.go`.